File: 151004/Tishalovich/rest_basics/publisher/src/main/java/by/bsuir/dc/publisher/entities/dtos/response/StoryResponseTo.java
package by.bsuir.dc.publisher.entities.dtos.response;

import org.springframework.data.redis.core.RedisHash;

import java.io.Serializable;
import java.util.Date;

@RedisHash("story")
public record StoryResponseTo(
        Long id,
        Long authorId,
        String title,
        String content,
        Date created,
        Date modified) implements Serializable {
}

File: 151004/Danilov/publisher/src/main/kotlin/com/danilovfa/model/story/StoryResponseTo.kt
package com.danilovfa.model.story

import kotlinx.serialization.Serializable

@Serializable
data class StoryResponseTo(
    val id: Long,
    val userId: Long,
    val title: String,
    val content: String,
    val created: Long,
    val modified: Long
)

fun Story.toResponse() =
    StoryResponseTo(
        id = id,
        userId = userId,
        title = title,
        content = content,
        created = created,
        modified = modified
    )
